src/systems/filecoin_vm/runtime/runtime.id

```diff
@@ -52,7 +51,19 @@ type Runtime interface {
     // Note: this should only be used for exceptional conditions, such as inconsistent state
     // values or precondition violations. Operations that may fail during normal execution
     // should use error return values, not call this method.
-    Abort(string)
+    Abort(errExitCode exitcode.ExitCode, msg string)
+
+    // Calls Abort with InvalidArguments_User.
+    AbortArgMsg(msg string)
+    AbortArg()
+
+    // Calls Abort with InconsistentState_User.
+    AbortStateMsg(msg string)
+    AbortState()
+
+    // Calls Abort with RuntimeAPIError.
+    // For internal use only (not in actor code).
```

Comment:
I agree -- this is a workaround to deal with Go's lack of generics. We currently have designated boilerplate at the top of each actor code file (say, for actor type T), which handles state updates specific to type T. Ideally, this boilerplate would not be duplicated, but rather would be type-generic and completely runtime-internal. Unfortunately, since we don't have generics, this means that either (a) the runtime needs to know about the internals of specific actors T; or (b) the actors need to know about internals of the runtime. The latter seems marginally more palatable since otherwise we would have to deal with circular dependencies in the code.

Reply:
How about (c) we don't clutter the spec with such boilerplate?

Reply:
Yes this

Reply:
Not sure what the concrete proposal is for (c), since the spec still needs to compile and accurately reflect the desired intent. But I'll take a look and see if I can simplify this API in a subsequent PR.